controller/controller.go

```diff
@@ -390,14 +390,18 @@ func ScaleUp(
 		"scaleUpLimit": scaleUpLimit,
 	})
 
-	configYAML, _ := models.NewConfigYAML(scheduler.YAML)
+	configYAML, err := models.NewConfigYAML(scheduler.YAML)
+	if err != nil {
+		return fmt.Errorf("failed to load scheduler configuration: %w", err)
+	}
 
-	existPendingPods, err := pendingOrFailedPods(config, redisClient, scheduler.Name, mr)
+	notReadyPods, err := hasNotReadyPods(config, redisClient, scheduler.Name, configYAML.PreventRoomsCreationWithError, mr)
```

Comment:
Sorry, me again... checking the correct English, we should call it as: `hasNonReadyPods` once `hasNotReadyPods` means that: there's no pod ready.